lib/popup.js

```diff
@@ -3,104 +3,91 @@ const measureDisplayEl = document.querySelector('.measure-display')
 const timeSpentEl = document.querySelector('.time-spent')
 const networkEl = document.querySelector('.network')
 const dateEl = document.querySelector('.date')
-let displayDate = 'Today'
-let displayHostname = ''
+const trackButton = document.getElementById('trackButton')
 
 const dates = {
   'Today': new Date(new Date().setHours(0, 0, 0, 0)),
   'Yesterday': new Date(new Date(new Date().setDate(new Date().getDate() - 1)).setHours(0, 0, 0, 0))
 }
 
-for (const date in dates) {
-  const dateItem = document.createElement('option')
-  dateItem.textContent = date
-  dateItem.value = date
-  dateEl.appendChild(dateItem)
-}
-
-chrome.storage.local.get({ hostNavigationStats: {}, config: {} }, results => {
-  // Return if the stats object is empty
-  if (Object.keys(results.hostNavigationStats).length === 0) {
-    return
-  }
-
-  const todayStats = results.hostNavigationStats[dates['Today']]
-  const sortedHostnames = Object.keys(todayStats).sort((a, b) => {
-    return todayStats[a] <= todayStats[b]
+chrome.storage.local.get({ hostNavigationStats: {}, config: {} }, res => {
+  chrome.tabs.query({active: true, currentWindow: true}, tabs => {
+    view(new URL(tabs[0].url).hostname, 'Today', res.config, res.hostNavigationStats)
   })
+})
 
-  if (sortedHostnames.length === 0) {
-    return
-  }
+function view (hostname, date, config, stats) {
+  const dayStats = stats[dates[date]]
+  // We take the five most visited websites, ordered by the time spent on them
+  const sortedHostnames = Object.keys(dayStats).sort((a, b) => dayStats[a] <= dayStats[b]).slice(0, 5)
+  const timeSpent = formatTime(getTimeSpent(stats, date, hostname))
 
-  // Setup listener for dropdown
-  networkEl.addEventListener('change', event => {
-    if (!event.target.value) {
-      alert('Please Select One')
-    } else {
-      displayHostname = event.target.value
-      estimateTime()
+  if (config.blacklist.includes(hostname)) { // this website is tracked
+    while (networkEl.firstChild) {
+      networkEl.remove(networkEl.firstChild)
     }
-  })
+    for (const site of sortedHostnames) {
+      const networkItem = document.createElement('option')
+      networkItem.textContent = site
+      networkItem.value = site
+
+      if (site === hostname) {
+        networkItem.selected = true
+      }
 
-  dateEl.addEventListener('change', event => {
-    if (!event.target.value) {
-      alert('Please Select One')
-    } else {
-      displayDate = event.target.value
-      estimateTime()
+      networkEl.appendChild(networkItem)
     }
-  })
 
-  function estimateTime () {
-    if (typeof results.hostNavigationStats[dates[displayDate]] === 'undefined' ||
-        typeof results.hostNavigationStats[dates[displayDate]][displayHostname] === 'undefined') {
-      timeSpentEl.textContent = 'No data'
-      return
+    while (dateEl.firstChild) {
+      dateEl.remove(dateEl.firstChild)
+    }
+    for (const possibleDate in dates) {
+      if (stats[dates[possibleDate]]) {
```

Comment:
OK, I can reproduce if I load the add-on in my personal Firefox, not the freshly installed one `web-ext run` gives you. I can see errors in the console which makes me thinks that it is because we store (and use) DOM nodes after the popup is closed (so these DOM nodes are destroyed). :thinking: